presentation/src/main/java/com/example/healthysalad/FragmentScheduler.kt

```diff
@@ -0,0 +1,33 @@
+package com.example.healthysalad
+
+
+class FragmentScheduler(private val currentState: String){
+    private val fragmentFlow : List<String> = listOf("base", "protein","crunchy", "dressing","soft")
+
+    fun getNextState() : String {
+        val pos = fragmentFlow.indexOf(currentState)
+        var next : String = fragmentFlow.last()
+        if (pos < fragmentFlow.size-1){ // position is 0-indexed while size is count
```

Comment:
should be `if (pos < fragmentFlow.size - 1) {`

Reply:
its the spacing right?